tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+        </data>
+    </outputs>
+
+    <tests>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <output name="out1" file="out1.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <param name="adapter_sequence1" value="ATCG"/>
+            <output name="out1" file="out_a.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <section name="umi_processing">
+                <param name="umi" value="true"/>
+                <param name="umi_loc" value="read1"/>
+                <param name="umi_len" value="8"/>
+            </section>
+            <output name="out1" file="out2.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="R1.fq"/>
+            <param name="single_paired_selector" value="single"/>
+            <section name="umi_processing">
+                <param name="umi" value="true"/>
+                <param name="umi_loc" value="read1"/>
+                <param name="umi_len" value="12"/>
+            </section>
+            <output name="out1" file="out3.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="bwa-mem-fastq1.fq"/>
+            <param name="in2" value="bwa-mem-fastq2.fq"/>
+            <param name="single_paired_selector" value="paired"/>
+            <output name="out1" file="out_bwa1.fq" ftype="fastq"/>
+            <output name="out2" file="out_bwa2.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="bwa-mem-fastq1.fq"/>
+            <param name="in2" value="bwa-mem-fastq2.fq"/>
+            <param name="single_paired_selector" value="paired"/>
+            <section name="umi_processing">
+                <param name="umi" value="true"/>
+                <param name="umi_loc" value="read1"/>
+                <param name="umi_len" value="8"/>
+            </section>
+            <output name="out1" file="out_bwa_umi_read1_1.fq" ftype="fastq"/>
+            <output name="out2" file="out_bwa_umi_read1_2.fq" ftype="fastq"/>
+        </test>
+        <test>
+            <param name="in1" value="bwa-mem-fastq1.fq"/>
+            <param name="in2" value="bwa-mem-fastq2.fq"/>
+            <param name="single_paired_selector" value="paired"/>
+            <section name="umi_processing">
+                <param name="umi" value="true"/>
+                <param name="umi_loc" value="read2"/>
+                <param name="umi_len" value="8"/>
+            </section>
+            <output name="out1" file="out_bwa_umi_read2_1.fq" ftype="fastq"/>
+            <output name="out2" file="out_bwa_umi_read2_2.fq" ftype="fastq"/>
+        </test>
```

Comment:
It would be nice to see some tests that test the parameters in some of the more advanced sections:

`base_correction_options`, `length_filtering_options`, `quality_filtering_options`, `polyG_tail_trimming`, `overrepresented_sequence_analysis`